lib/plug/adapters/test/conn.ex

```diff
@@ -148,9 +148,27 @@ defmodule Plug.Adapters.Test.Conn do
   defp body_or_params(params, query, headers, method)
        when is_map(params) and method in ["GET", "HEAD"] do
     params = stringify_params(params, &to_string/1)
-    params = Map.merge(Plug.Conn.Query.decode(query), params)
 
-    {"", nil, params, headers}
+    query = Plug.Conn.Query.decode(query)
+    params = Map.merge(query, params)
+
+    query =
+      params
+      |> Enum.reject(fn
+        {_, v} when is_function(v) ->
+          true
+
+        {_, %{__struct__: _mod}} ->
+          true
+
+        _ ->
+          false
+      end)
```

Comment:
Why do we reject those? I think everything we pass needs to be convertible to a string. Otherwise, it is fine to crash.

Reply:
Because apparently (as I've just learned!), it is possible to pass in function and also structs, as you can see below:

https://github.com/elixir-plug/plug/blob/c1790e5e692b87dd2042c9288abaebb93b5f6122/test/plug/adapters/test/conn_test.exs#L78

https://github.com/elixir-plug/plug/blob/c1790e5e692b87dd2042c9288abaebb93b5f6122/test/plug/adapters/test/conn_test.exs#L66

So I figured we'd need to drop those

Reply:
Both of those tests are invalid, so we can remove them. :)

Reply:
Interesting! I've gone ahead and removed them and simplified the code now, looks much better